pkg/scheduler/factory/factory.go

```diff
@@ -1055,6 +1058,116 @@ func (c *configFactory) invalidateCachedPredicatesOnNodeUpdate(newNode *v1.Node,
 	}
 }
 
+func nodeSchedulingPropertiesChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if nodeAllocatableChanged(newNode, oldNode) {
+		glog.V(4).Infof("Allocatable resource of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeLabelsChanged(newNode, oldNode) {
+		glog.V(4).Infof("Labels of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeTaintsChanged(newNode, oldNode) {
+		glog.V(4).Infof("Taints of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeConditionsChanged(newNode, oldNode) {
+		glog.V(4).Infof("Conditions of node %s changed", newNode.Name)
+		return true
+	}
+	if newNode.Spec.Unschedulable != oldNode.Spec.Unschedulable && newNode.Spec.Unschedulable == false {
+		glog.V(4).Infof("Node %s changed to schedulable", newNode.Name)
+		return true
+	}
+	return false
+}
+
+func nodeAllocatableChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	oldAllocatable := oldNode.Status.Allocatable
+	newAllocatable := newNode.Status.Allocatable
+	if reflect.DeepEqual(oldAllocatable, newAllocatable) {
+		return false
+	}
+	for resource, newValue := range newAllocatable {
```

Comment:
What you have done makes sense, but in order to be quicker in checking node changes and also to be conservative, I would return "true" as long allocatables are changed, no matter whether they are reduced or increased. In other words, remove this for loop and just return true.

Reply:
Done